client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -163,7 +163,7 @@ var CheckoutThankYou = React.createClass( {
 	 * @returns {*[]} an array of varying size with the component instance, then an optional purchase object possibly followed by a domain name
 	 */
 	getComponentAndPrimaryPurchaseAndDomain: function() {
-		if ( this.isDataLoaded() ) {
+		if ( this.isDataLoaded() && this.props.receiptId ) {
```

Comment:
What is the benefit of checking `props.receipt.data.purchases` instead of `props.receipt.hasLoadedFromServer`? They should both be truthy at the same time, and the latter exists to prevent the need to do presence checks like this.

Either way, we need two booleans (or maybe a single string) to determine which of the three states to render.